zebra-state/src/service/finalized_state/zebra_db/shielded.rs

```diff
@@ -355,6 +436,92 @@ impl DiskWriteBatch {
             self.zs_insert(&orchard_tree_cf, height, trees.orchard);
         }
 
+        if let Some(subtree) = trees.sapling_subtree {
+            let (subtree_end, subtree_start) = subtree.bounds();
+
+            let start_height = if subtree_end.index == 0 {
+                let (start_height, _) = zebra_db
+                    .sapling_tree_by_height_range(Height(1)..)
+                    .next()
+                    .expect("must have sapling notes on disk");
+
+                start_height
```

Comment:
I think this might be wrong. We need to check at which height the subtrees actually start.

Reply:
The idea here is that if it's the first sapling subtree, its start height is the first height with a unique sapling note commitment treestate?

Reply:
Oh, I forgot we deduplicated the trees. LOL

Reply:
Added comments in https://github.com/ZcashFoundation/zebra/pull/7334/commits/aea9320805ed46f7fd58866644b792267592ea7b.